src/plugins/intel_gna/tests/unit/transformations/gna_insert_copy_layer.cpp

```diff
@@ -1519,35 +1486,90 @@ TEST(TransformationTests, InsertCopyLayerNFLvsFLSubgraphTestt) {
+        auto split = ngraph::builder::makeSplit(params, ov::element::i64, 1, axis);
         auto reshape = ov::op::util::reshapeTo(split->output(0), shape);
-        auto const_value = ngraph::builder::makeConstant(ngraph::element::i64, shape, std::vector<size_t>{1});
-        auto concat = std::make_shared<ngraph::opset8::Concat>(ngraph::OutputVector{reshape, const_value}, axis);
-        auto result = std::make_shared<ngraph::opset8::Result>(concat);
-        func =
-            std::make_shared<ngraph::Function>(ngraph::ResultVector{result}, ngraph::ParameterVector{params}, "Concat");
+        auto const_value = ngraph::builder::makeConstant(ov::element::i64, shape, std::vector<size_t>{1});
+        auto concat = std::make_shared<ov::opset11::Concat>(ov::OutputVector{reshape, const_value}, axis);
+        auto result = std::make_shared<ov::opset11::Result>(concat);
+        func = std::make_shared<ov::Model>(ov::ResultVector{result}, ov::ParameterVector{params}, "Concat");
     }
     {
-        auto params = std::make_shared<ngraph::opset8::Parameter>(ngraph::element::i64, input_shape);
-        auto split = ngraph::builder::makeSplit(params, ngraph::element::i64, 1, axis);
+        auto params = std::make_shared<Parameter>(ov::element::i64, input_shape);
+        auto split = ngraph::builder::makeSplit(params, ov::element::i64, 1, axis);
         auto reshape = ov::op::util::reshapeTo(split->output(0), shape);
         auto copy = std::make_shared<ov::intel_gna::op::Copy>(reshape);
-        auto const_value = ngraph::builder::makeConstant(ngraph::element::i64, shape, std::vector<size_t>{1});
-        auto concat = std::make_shared<ngraph::opset8::Concat>(ngraph::OutputVector{copy, const_value}, axis);
+        auto const_value = ngraph::builder::makeConstant(ov::element::i64, shape, std::vector<size_t>{1});
+        auto concat = std::make_shared<ov::opset11::Concat>(ov::OutputVector{copy, const_value}, axis);
 
-        auto result = std::make_shared<ngraph::opset8::Result>(concat);
-        ref_func =
-            std::make_shared<ngraph::Function>(ngraph::ResultVector{result}, ngraph::ParameterVector{params}, "Concat");
+        auto result = std::make_shared<ov::opset11::Result>(concat);
+        ref_func = std::make_shared<ov::Model>(ov::ResultVector{result}, ov::ParameterVector{params}, "Concat");
+    }
+
+    ov::pass::Manager m;
+    m.register_pass<ov::pass::InitNodeInfo>();
+    m.register_pass<ov::intel_gna::pass::InsertCopyBeforeConcatLayer>();
+    m.run_passes(func);
+
+    ASSERT_NO_THROW(check_rt_info(func));
+
+    auto result = compare_functions(func, ref_func);
+    ASSERT_TRUE(result.first);
+}
+
+// [Parameter]              [Parameter]
+//     |                         |
+//  [Slice]                   [Slice]
+//     |           =>            |
+//   [FQ]                      [FQ]
+//     |                         |
+// [NonFunc] [Constant]      [NonFunc]
```

Comment:
Why NonFunc instead of Reshape?

Reply:
Reshape is one of non-functional layers.